lib/controller.js

```diff
@@ -229,18 +229,24 @@ function check_directory_exists(input, directory) {
 
 function check_file_exists(input, file) {
     let file_object = Gio.File.new_for_path(resolve_path(file));
-    try {
-        if (file_object.query_info('standard::type',
-                                   Gio.FileQueryInfoFlags.NONE,
-                                   null).get_file_type() == Gio.FileType.REGULAR) {
-            return ['success', []];
-        } else {
-            return ['failure', []];
+    let retries = 5;
+
+    while (retries--) {
+        try {
+            if (file_object.query_info('standard::type',
+                                       Gio.FileQueryInfoFlags.NONE,
+                                       null).get_file_type() == Gio.FileType.REGULAR) {
+                return ['success', []];
+            } else {
+                return ['failure', []];
+            }
+        } catch (e) {
+            logError(e, 'Retrying, ' + retries + ' remaining');
+            GLib.usleep(GLib.USEC_PER_SEC / 100);
```

Comment:
This is going to lock the entire process, which is not ideal -- you may avoid this whole thing perhaps with my suggestion in the PR.